optimized_ingestion/stages/detection_estimation/detection_estimation.py

```diff
@@ -109,38 +110,81 @@ def compute_geo_info(self):
     def compute_priority(self):
         self.priority = self.segment_area_2d / self.distance
 
-    def generate_single_sample_action(self, view_distance: float = 50.):
-        """Generate a sample plan for the given detection of a single car
-
-        Condition 1: detected car is driving towards ego, opposite direction
-        Condition 2: detected car driving along the same direction as ego
-        Condition 3: detected car and ego are driving into each other
-                    i.e. one to north, one from west to it
-        Condition 4: detected car and ego are driving away from each other
-                    i.e. one to north, one from it to east
-        Return: a list of actions
-        """
-        sample_action_alg = get_sample_action_alg(self.relative_direction)
-        if sample_action_alg is not None:
-            return self.priority, sample_action_alg(self, view_distance)
-        return self.priority, None
+    def get_car_exits_segment_action(self):
+        current_time = self.timestamp
+        car_loc = self.car_loc3d
+        exit_time, exit_point = time_to_exit_current_segment(self, current_time, car_loc)
+        if exit_time is None or exit_point is None:
+            return None
+        return Action(current_time, exit_time, start_loc=car_loc,
+                      end_loc=exit_point, action_type=CAR_EXIT_SEGMENT,
+                      target_obj_id=self.detection_id,
+                      target_obj_bbox=self.car_bbox2d)
```

Comment:
I'm breaking the abstraction barrier here and directly compute the car exits segment time here.